### DISCUSSION TITLE
LazyVim is slow, Can I do something about it?

### DISCUSSION THREAD

**Discussion by aizaz-shahid** [LazyVim/LazyVim#3427] [Q&A] [answered]
I recently switched to Neovim and used TJ's Kickstart for easy configuration. Now, I have a customised setup. However, it loads files very slowly the first time; opening a 1500-line Python file takes about five seconds. I attempted to lazy load several plugins using events, but it didn't improve the speed. Then, I tried NvChad, which opened the same file almost instantly. I'd prefer not to reconfigure NvChad from scratch. Is there a way to speed up Lazyvim?

Note: I tested the starter config to check if the issue was with my configuration, but the loading speed was still slow.

Here is my config: https://github.com/aizaz-shahid/neovim

Details of my system:
MacBook Pro
16-inch, 2019
Processor
2.6 GHz 6-Core Intel Core
i7
Graphics
AMD Radeon Pro 5300M
4 GB
Intel UHD Graphics 630
1536 MB
Memory
16 GB 2667 MHz DDR4
macOS Sonoma 14.5

**Comment by dpetka2001** [comment]
From your dotfiles it doesn't look like you're using LazyVim. You have a custom setup with `lazy.nvim` as your package manager.

**Reply by aizaz-shahid** [reply]
Yes, that is my custom config that I created using Kickstart, as I explained above. But like I said lazyvim was the same using the starter config. And NvChad is super fast for some reason.

**Comment by aizaz-shahid** [comment, marked as answer]
Finally figured out the root cause for me: It was loading python. Adding the below lines solved it for me:
```
let g:python_host_prog  = '/usr/bin/python2'
let g:python3_host_prog = '/usr/bin/python3'
```

**Reply by dpetka2001** [reply]
I just cloned the tensorflow repository and opened `configure.py` which is 1523 lines and it took at most 1 sec to open it. Screencast following 

[Screencast 2024-06-02 22:16:10.webm](https://github.com/LazyVim/LazyVim/assets/12776461/a28d8d77-6d21-43c7-8f32-be894c0701a8)

This is default LazyVim starter with the Python Extra enabled.

**Reply by mattdkerr** [reply]
First I went and removed a number of plugins I wasn't going to use immediately, for troubleshooting, then I made a symbolic link in Windows under Git Bash environment: 
`ln -s <path to python3> /usr/bin/python3` 
`ln -s <path to python> /usr/bin/python`
then start-up was very fast
